website/static/js/fangorn.js

```diff
@@ -2605,13 +2612,13 @@ function displayMoveStats(tb) {
 function doSyncMove(tb, provider){
     var cache = tb.syncFileMoveCache && tb.syncFileMoveCache[provider];
     var itemData;
-    if (cache.conflicts && cache.conflicts.length > 0) {
+    if (cache.conflicts && cache.conflicts.length > 0 && !MOVE_INTERVAL) {
         itemData = cache.conflicts.pop();
         displayConflict(tb, itemData.item, itemData.folder, doItemOp.bind(tb, copyMode === 'move' ? OPERATIONS.MOVE : OPERATIONS.COPY, itemData.folder, itemData.item, undefined));
     } else if (cache.ready && cache.ready.length > 0) {
         itemData = cache.ready.pop();
         doItemOp.call(tb, copyMode === 'move' ? OPERATIONS.MOVE : OPERATIONS.COPY, itemData.folder, itemData.item, undefined, 'replace');
-    } else {
+    } else if (!MOVE_INTERVAL && !cache.ready && !cache.conflicts) {
```

Comment:
This seems significantly more constrained than it was before, which is fine, but I'm not familiar enough with the code otherwise. It seems to make sense, but this is not particularly tested code, and what used to be a fall-through condition now has a bunch of constraints, so I just want to be sure there are no missing cases in the new setup.

For readability, it might make sense to separate out the `if` logic into a separate function that makes something more declarative about the state. so add some constants at the top for a person-readable explanation of the state, then have the function read those conditions and return one of those constants which are then used in the if statement. Hopefully this code won't live too much longer, but that should be a quick and safe change.

Reply:
I tried to make the different states here more explicit, hopefully this is a little better, but I'm limited in what I can do without bigger changes.